src/components/elementRenderer/elementRenderer.tsx

```diff
@@ -11,13 +11,31 @@ interface ElementRendererProps {
 const ElementRenderer = (props: ElementRendererProps) => {
   const MaybeElement = props.supportedElements[props.message.format]
 
+  function extractFormat(updateFormat: string) {
+    const lettersInUpdate = 'update'.length
+    const format = updateFormat
+      .substring(0, updateFormat.length - lettersInUpdate)
+      .trim()
+
+    return format
+  }
+
+  const updateMessages = props.message.threadMessages?.filter(
+    (threadMessage) =>
+      threadMessage.sender === props.message.sender &&
```

Comment:
Should we avoid checking sender here? One component may get updates from two agents.

Reply:
It could. What do you think @Shiti ?

Reply:
could you give some examples where this would be the case - i.e., real-time updates to single component with data from two different sources ?

Reply:
I suppose if results from another agent would need to be reflected in another agent's message, that can be communicated on the backend and the original agent sender can send the update accordingly.

Reply:
I think that it can get confusing from user point of view and introduce overwriting of data. Please share any use-cases or example scenarios where this is expected or required